jobboard/tests/search_bar_tests.py

```diff
@@ -0,0 +1,108 @@
+import pytest
+from jobboard.models import Job
+
+# The tests use the data that is already in the db from the migration files
+
+
+@pytest.mark.django_db
+def test_search_bar_empty_search(client, example_student_a):
+    client.force_login(example_student_a.user)
+    search_bar_data = {"searched_name": "", "searched_city": ""}
+    response = client.post("/", data=search_bar_data)
+    assert response.context["jobs_to_display"] == set()
+
+
+@pytest.mark.django_db
+def test_search_bar_company(client, example_student_a):
+    job_a = Job.objects.get(title="job_a")
+    job_b = Job.objects.get(title="job_b")
+    job_c = Job.objects.get(title="job_c")
+    job_d = Job.objects.get(title="job_d")
+    job_e = Job.objects.get(title="job_e")
+    client.force_login(example_student_a.user)
+    # searches for jobs at 'example_company_b'
+    search_bar_data = {"searched_name": "Example_Company_B", "searched_city": ""}
+    response = client.post("/", data=search_bar_data)
+    assert response.context["jobs_to_display"] == {job_b, job_d}
+    # searches for jobs at 'example_company_a'
+    search_bar_data = {"searched_name": "Example_Company_a", "searched_city": ""}
+    response = client.post("/", data=search_bar_data)
+    assert response.context["jobs_to_display"] == {job_a, job_c, job_e}
+
+
+@pytest.mark.django_db
+def test_search_bar_keyword(client, example_student_a):
+    job_a = Job.objects.get(title="job_a")
+    job_b = Job.objects.get(title="job_b")
+    job_c = Job.objects.get(title="job_c")
+    job_d = Job.objects.get(title="job_d")
+    job_e = Job.objects.get(title="job_e")
+    client.force_login(example_student_a.user)
+    # searches for jobs with keyword 'software'
+    search_bar_data = {"searched_name": "software", "searched_city": ""}
+    response = client.post("/", data=search_bar_data)
+    assert response.context["jobs_to_display"] == {job_a, job_b, job_e}
+    # searches for jobs with keyword 'product'
+    search_bar_data = {"searched_name": "product", "searched_city": ""}
+    response = client.post("/", data=search_bar_data)
```

Comment:
I remember Barak said that we should try to keep each test with one HTTP request only. So maybe testing a search with only one keyword is enough. (Same goes for the other tests).

Reply:
Done